### PR TITLE
feat(docker): Add docker build workflow

### PR DESCRIPTION
**Pull request by AlexAxthelm**
Add workflow that builds multiple architectures of a docker image in parallel, and merges them into a single tagged image.

No Assosciated issue, but implementing a feature that we have parts of in multiple repos.

Relates to https://github.com/RMI-PACTA/workflow.pacta/issues/20 
Relates to https://github.com/RMI-PACTA/workflow.pacta/issues/26 
Relates to https://github.com/RMI-PACTA/workflow.pacta/issues/29
Relates to https://github.com/RMI-PACTA/workflow.pacta/issues/32

Closes #56

TODO:
* [ ] Update documentation

### CODE REVIEW

**Comment by AlexAxthelm** [comment]
@cjyetman @jdhoffa I think this is ready for review.

I've set up a temporary repo `workflow.pacta.actions` that I've been using while I develop this PR (don't look too closely at the git history), but in its current form, you'll see a PR against what is the current `main` on `workflow.pacta` [`77f6ee0`](https://github.com/RMI-PACTA/workflow.pacta/commit/77f6ee025d06265e7ab412f01497773f1d812701), and has basically stripped out all the workflows and replaced them with this (I've probably stripped too far, since I've only left the PR actions, not the push ones).

There are some minor changes to the dockerfile there to enable building R packages on multiple platforms, which I stole from @cjyetman in https://github.com/RMI-PACTA/workflow.pacta/pull/31

Take a look at that repo, check out the packages, let me know what you think.

As far as reviewing this, I'm so deep in it at this point, I'm not sure what's clear and what's not, so if you need clarification on anything, flag it, and I'll document as needed.

**Comment by jdhoffa** [comment]
This is incredibly cool. 

As with most "action" PRs, I don't have a great way to test this, other than trying to use it once it's merged?

I will give it a shot in `workflow.scenario.preparation` shortly.

File: .github/workflows/docker-build-multiarch.yml [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+    inputs:
+      image-name:
+        description: image to build, including registry (only ghcr.io supported)
+        required: true
+        type: string
+      registry:
+        description: registry to push image to (only default ghcr.io is currently supported)
+        required: false
+        type: string
+        default: ghcr.io
```
Comment (jdhoffa) [inline .github/workflows/docker-build-multiarch.yml:18]: NIT: Odd that `image_name` requires the registry to be specified, but then you specify `registry` right after?

Maybe double-check this? I can imagine accidents if someone specifies an `image_name` registry and `registry` parameters that don't agree.

**Review by jdhoffa — CHANGES_REQUESTED** [review]
A NIT comment so far. 

I will look into the `workflow.pacta.actions` repo to see the output I suppose

**Comment by jdhoffa** [comment]
Can you link the PR you are talking about that you "stripped too far"?

**Comment by AlexAxthelm** [comment]
> Can you link the PR you are talking about that you "stripped too far"?

Specifically [this commit](https://github.com/RMI-PACTA/workflow.pacta.actions/pull/2/commits/22b7d2d7e770e8aec6945cd284d593ba00164cef) in the demo repo, where I get rid of all the action triggers that aren't PR

File: .github/workflows/docker-build-multiarch.yml (hunk 1 above)
Comment (AlexAxthelm) [reply .github/workflows/docker-build-multiarch.yml:18]: Good catch. Sorting that out now.

Comment (AlexAxthelm) [reply .github/workflows/docker-build-multiarch.yml]: `image-name` no longer requires registry, but does require the username for namespacing within the registry (e.g. `rocker/r-ver` or `RMI-PACTA/workflow.pacta`)

**Review by jdhoffa — APPROVED** [review]
All I can do really is very cautiously approve this, and likely try to test the action in the wild on a relevant repo!

**Pull request merged**